### PR TITLE
chore: enable unicorn/prefer-string-replace-all

### PR DESCRIPTION
**Pull request by abrahamguo**
## PR Checklist

- [x] Addresses an existing open issue: fixes part of #9623 
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Enable `unicorn/prefer-string-replace-all` ([docs](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-replace-all.md)). 100% autofixable.

### CODE REVIEW

File: packages/ast-spec/tests/fixtures.test.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ const FIXTURES: readonly Fixture[] = [...VALID_FIXTURES, ...ERROR_FIXTURES].map(
       isError: /[\\/]_error_[\\/]/.test(absolute),
       isJSX: ext.endsWith('x'),
       name,
-      relative: path.relative(SRC_DIR, absolute).replace(/\\/g, '/'),
+      relative: path.relative(SRC_DIR, absolute).replaceAll('\\', '/'),
```
Comment (JoshuaKGoldberg) [inline packages/ast-spec/tests/fixtures.test.ts:80]: 🤔 this new version is more characters and IMO equally clear. I'm -0.1.

Comment (abrahamguo) [reply packages/ast-spec/tests/fixtures.test.ts:80]: In this specific instance, or in the whole PR?

Comment (Josh-Cena) [reply packages/ast-spec/tests/fixtures.test.ts:80]: I'm +1. Regex escaping is too footguny. We should prioritize string replacement whenever possible.

Comment (JoshuaKGoldberg) [reply packages/ast-spec/tests/fixtures.test.ts:80]: I meant kind of halfway between this instance & the whole PR. This instance and the ones like it...

...but yeah, Josh brings up a good point on the footguns. I'm ok with the average being +0.9.

**Review by kirkwaiblinger — APPROVED** [review]

**Pull request merged**